GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -376,8 +462,14 @@ private void BuildOrderedRowCache(RevisionGraphRevision[] orderedNodesCache, int
                     segments = new List<RevisionGraphSegment>(previousRevisionGraphRow.Segments.Count + revisionStartSegments.Length);
 
                     // Loop through all segments that do not end in the previous row
-                    foreach (RevisionGraphSegment segment in previousRevisionGraphRow.Segments.Where(s => s.Parent != previousRevisionGraphRow.Revision))
+                    for (int i = 0; i < previousRevisionGraphRow.Segments.Count; i++)
```

Comment:
Does the compiler realize that `previousRevisionGraphRow.Segments.Count` does not change?
Otherwise
```suggestion
                    for (int i = 0, count = previousRevisionGraphRow.Segments.Count; i < count; ++i)
```

Reply:
I cannot see how the compile can determine that, using a variable.